plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TestCheckpointEntryIterator.java

```diff
@@ -233,6 +241,58 @@ public void testReadAddEntries()
                         Optional.empty()));
     }
 
+    @Test
+    public void testReadAddEntriesPartitionPruning()
+            throws Exception
+    {
+        String checkpoint = "statistics/partition_values_parsed/_delta_log/00000000000000000002.checkpoint.parquet";
+        URI checkpointUri = getResource(checkpoint).toURI();
+
+        DeltaLakeColumnHandle stringPartField = new DeltaLakeColumnHandle(
+                "string_part",
+                VARCHAR,
+                OptionalInt.empty(),
+                "string_part",
+                VARCHAR,
+                REGULAR,
+                Optional.empty());
+
+        DeltaLakeColumnHandle intPartField = new DeltaLakeColumnHandle(
+                "int_part",
+                BIGINT,
+                OptionalInt.empty(),
+                "int_part",
+                BIGINT,
+                REGULAR,
+                Optional.empty());
+
+        CheckpointEntryIterator checkpointEntryIterator = createCheckpointEntryIterator(
+                checkpointUri,
+                ImmutableSet.of(ADD),
+                Optional.of(readMetadataEntry(checkpointUri)),
+                Optional.of(readProtocolEntry(checkpointUri)),
+                TupleDomain.withColumnDomains(ImmutableMap.of(intPartField, singleValue(BIGINT, 10L), stringPartField, singleValue(VARCHAR, utf8Slice("part1")))));
+        List<DeltaLakeTransactionLogEntry> entries = ImmutableList.copyOf(checkpointEntryIterator);
+
+        assertThat(entries).hasSize(2);
```

Comment:
Shouldn't we have here only 1 entry?

We should have also 

```
        assertThat(checkpointEntryIterator.getCompletedPositions().orElseThrow()).isEqualTo(....);
```

Reply:
When doing `buildAddEntry` check whether the `partitionValues` / `partitionValues_parsed` match the `partitionConstraint` and return `null` if not matching.

https://github.com/trinodb/trino/blob/10fdc951f03f08cc366bc9f748de92ee9c0d3fa6/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java#L212-L213